### PR TITLE
feat(workspace/app_policy_group): add resource to manage  a policy group

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
Add new resource to manage a Workspace APP policy group.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource.
2. add corresponding document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
 make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccAppPolicyGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccAppPolicyGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccAppPolicyGroup_basic
=== PAUSE TestAccAppPolicyGroup_basic
=== CONT  TestAccAppPolicyGroup_basic
--- PASS: TestAccAppPolicyGroup_basic (69.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 69.277s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
  image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o workspace -f TestAccAppPolicyGroup_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccAppPolicyGroup_basic -timeout 360m -parallel 10
=== RUN   TestAccAppPolicyGroup_basic
=== PAUSE TestAccAppPolicyGroup_basic
=== CONT  TestAccAppPolicyGroup_basic
--- PASS: TestAccAppPolicyGroup_basic (120.53s)
PASS
coverage: 10.9% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 121.304s        coverage: 10.9% of statements in ./huaweicloud/services/workspace

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_app_policy_group.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("appstream", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace APP client: %s", err)
+	}
+
+	policyGroup, err := GetAppGroupPolicy(client, d.Get("name").(string), d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retring Workspace APP policy group")
```
Comment (ruwenqiang123) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_app_policy_group.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error retrieving Workspace APP policy group")
```

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_app_policy_group.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+		return diag.Errorf("error creating Workspace APP policy group: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	policyGroupId := utils.PathSearch("id", respBody, "")
+	d.SetId(policyGroupId.(string))
```
Comment (ruwenqiang123) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_app_policy_group.go]:
```suggestion
	policyGroupId := utils.PathSearch("id", respBody, "").(string)
	if policyGroupId == "" {
		return diag.Errorf("error creating Workspace APP policy group: ID is not found in API response")
	}
	d.SetId(policyGroupId)
```

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by Alice1319** [comment]
LGTM

**Comment by dengyali0125** [comment]
LGTM

**Comment by dengyali0125** [comment]
LGTM

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_app_policy_group_test.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    type = "APPGROUP"
+  }
+
+  policies = jsonencode({
+    "client" : {
+      "automatic_reconnection_interval" : 10,
+      "session_persistence_time" : 120,
+      "forbid_screen_capture" : true
+    },
+  })
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_app_policy_group_test.go]:
```suggestion
    "client": {
      "automatic_reconnection_interval" : 10,
      "session_persistence_time" : 120,
      "forbid_screen_capture" : true
    }
  })
```

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccAppPolicyGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccAppPolicyGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccAppPolicyGroup_basic
=== PAUSE TestAccAppPolicyGroup_basic
=== CONT  TestAccAppPolicyGroup_basic
--- PASS: TestAccAppPolicyGroup_basic (62.84s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 62.891s

**Pull request merged**